tools/Customization/DevHome.Customization/Views/AddRepositoriesView.xaml

```diff
@@ -0,0 +1,75 @@
+<?xml version="1.0" encoding="utf-8"?>
+<UserControl
+    x:Class="DevHome.Customization.Views.AddRepositoriesView"
+    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
+    xmlns:ctControls="using:CommunityToolkit.WinUI.Controls" 
+    xmlns:viewmodels="using:DevHome.Customization.ViewModels"
+    mc:Ignorable="d">
+
+    <StackPanel>
+        <TextBlock 
+            x:Uid="AddRepositoriesTextBlock"
+            Style="{StaticResource SettingsSectionHeaderTextBlockStyle}"/>
+        <ctControls:SettingsCard
+            x:Uid="AddRepositoriesCard"
+            Grid.Column="1">
+            <Button 
+                x:Uid="AddFolderButton" 
+                Style="{ThemeResource AccentButtonStyle}" 
+                Command="{x:Bind ViewModel.AddFolderClickCommand}" />
+        </ctControls:SettingsCard>
+        <ItemsRepeater 
+            Name="ItemsRepeaterForAllRepoPaths" 
+            ItemsSource="{x:Bind ViewModel.TrackedRepositories, Mode=OneWay}"
+            Margin="{ThemeResource SettingsCardMargin}">
+            <ItemsRepeater.ItemTemplate>
+                <DataTemplate>
+                    <ctControls:SettingsCard 
+                        Header="{Binding RepositoryRootPath}"
+                        Margin="0">
+                        <ctControls:SettingsCard.HeaderIcon>
+                            <FontIcon 
+                                FontFamily="{StaticResource SymbolThemeFontFamily}" 
+                                Glyph="&#xf12b;"/>
+                        </ctControls:SettingsCard.HeaderIcon>
+                        <StackPanel 
+                            Orientation="Horizontal">
+                            <DropDownButton 
+                                Content="{Binding SourceControlProviderDisplayName}">
+                                <DropDownButton.Flyout>
+                                    <MenuFlyout 
+                                        Placement="Bottom">
+                                        <!-- WinUI does not currently allow populating MenuFlyoutItems dynamically within a DataTemplate. -->
```

Comment:
Please file an issue/bug to track this. We only have the one provider at the moment, so this shouldn't block some testing/selfhosting. But let's not forget to come back to this and fix it.

Reply:
Adding this as a potential fix.  Using an MSBuild task to iterate through all the extensions and populate XAML before it is built.  It would be difficult though.  Just an idea.